rest-api-spec/src/main/resources/rest-api-spec/api/bulk.json

```diff
@@ -90,7 +90,8 @@
       }
     },
     "body":{
-      "description":"The operation definition and data (action-data pairs), separated by newlines",
+      "type":"list",
+      "description":"The operation definition and data (action-data pairs), separated by newlines E.g. 'action\ndata\naction\naction\ndata'. In client libraries (like NodeJS) action-data pair objects should be used as array members; data is optional according to action type. E.g. '[action,data,action,action,data]'",
```

Comment:
IMHO a lot of the clarifications here are already explained in the reference documentation for bulk: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html (e.g. NDJSON structure etc.)

Reply:
I concur with with @dliappis, the description is serves as short description what gets send not necessarily documenting how.

Reply:
It is okay but if you check this: https://github.com/elastic/elasticsearch-js/pull/1040 the elasticsearch-js reference is auto-generated from these descriptions. In api reference bulk request's body argument has explanation like "seperated by new lines json". This is meanless for NodeJS. So i created a PR for that repo but @delvedor point it, that is an auto-generated file. So i think there are things to fix in elasticsearch-js api reference. Bulk request body argument type called object in there and i saw lots of complications about this on web. This is just clarification.

Reply:
I agree with @dliappis and @Mpdreamz. This description should be short and sweet, a link to the official documentation is provided as well.
Probably what we can do in the js client is to add another link to the [examples](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/bulk_examples.html) section.